### PR TITLE
[WIP] Feature/style changes

### PR DESCRIPTION
**Pull request by nburkley**
This is a WIP PR for the updated styling of the Espi-DNI site. Designs are available in Zeplin.

Hoping @eschaefer can tidy up my hacky css.

### CODE REVIEW

**Comment by nburkley** [comment]
Actually, if anyone from @edenspiekermann/css has a few mins quick code review might help me with what I shouldn't be doing. I might not have time to implement the changes before vacation but it's probably good to learn anyway.

File: web/static/css/app.css [hunk 1]
```diff
@@ -119,6 +119,10 @@ a {
   display: none;
 }
 
+header {
```
Comment (martzoukos) [inline web/static/css/app.css]: Can you use a class for that?

File: web/static/css/app.css [hunk 2]
```diff
@@ -150,6 +154,8 @@ a {
 
 .hero__logo {
   width: 200px;
+  -webkit-filter: drop-shadow(-2px -2px 5px #656c70);
+  filter: drop-shadow(-2px -2px 5px #656c70);
```
Comment (martzoukos) [inline web/static/css/app.css:158]: I guess asking for an autoprefixer precompiler is out of the PR's scope?

File: web/static/css/app.css [hunk 3]
```diff
@@ -165,14 +171,34 @@ a {
...
-  background-color: #47B9B4;
+  background-color: white;
   padding: 20px;
 }
 
 .bottom__title {
   font-size: 35px;
 }
 
+.bottom__back-link {
```
Comment (martzoukos) [inline web/static/css/app.css:182]: Again, probably out of this PR's scope but in general, you should have separate files for each BEM object. It'll make things easier to maintain.

File: web/static/css/app.css [hunk 4]
```diff
@@ -165,14 +171,34 @@ a {
...
+  font-size: 16px;
+  font-weight: 700;
+  color: #9b9b9b;
+  padding: 10px;
+}
+
+.bottom__back-link a {
+  color: #9b9b9b;
```
Comment (martzoukos) [inline web/static/css/app.css:190]: And the fact that you use the same colour in many places is an extra reason why you should use a preprocessor like Sass.

File: web/static/css/app.css [hunk 5]
```diff
@@ -265,6 +292,31 @@ footer {
...
+}
+
+.team-management h2 {
+  padding: 40px 0 20px 0;
+}
+
+.preferences {
+}
+
+.preferences__input--inline {
```
Comment (martzoukos) [inline web/static/css/app.css]: Is there a `.preferences__input` anywhere?

File: web/static/css/app.css [hunk 6]
```diff
@@ -313,6 +360,57 @@ footer {
...
+}
+
+.setup__header {
+  text-align: center;
+  line-height: 40px;
+  background-color: #31b0e2;
+  color: white;
+}
+
+.setup__header h2 {
```
Comment (martzoukos) [inline web/static/css/app.css]: Don't style elements, use classes.

File: web/templates/article/index.html.eex [hunk 7]
```diff
@@ -5,49 +5,43 @@
...
 
-            <td class="table__cell">
-              <%= link "Edit", to: article_path(@conn, :edit, article), class: "btn btn-default btn-xs" %>
-              <%= link "Delete", to: article_path(@conn, :delete, article), method: :delete, data: [confirm: "Are you sure?"] %>
-            </td>
-          </tr>
-        <% end %>
-      </tbody>
-    </table>
+                <td class="table__cell articles__action">
```
Comment (martzoukos) [inline web/templates/article/index.html.eex:35]: Try not to mix different classes in the same element. Here it should either be a variation of `table__cell` or use a different container (like a span) inside with the class `articles__action`.

**Review by martzoukos — APPROVED** [review]

File: web/static/css/app.css (hunk 1 above)
Comment (nburkley) [reply web/static/css/app.css]: cool, didn't realise I should avoid targeting elements

File: web/static/css/app.css (hunk 2 above)
Comment (nburkley) [reply web/static/css/app.css:158]: yes, yes it...

File: web/static/css/app.css (hunk 3 above)
Comment (nburkley) [reply web/static/css/app.css:182]: yeah, this is how @eschaefer set things up for now, as it's just one or two pages

File: web/static/css/app.css (hunk 1 above)
Comment (martzoukos) [reply web/static/css/app.css]: That's a good link to study about CSS specificity and why all these weird rules with BEM and "no element styles": https://css-tricks.com/strategies-keeping-css-specificity-low/

File: web/static/css/app.css (hunk 5 above)
Comment (nburkley) [reply web/static/css/app.css]: there's not, so I guess this should just be `preferences__input`, no need to set the type unless it's necessary?

File: web/static/css/app.css (hunk 6 above)
Comment (nburkley) [reply web/static/css/app.css]: gotcha

File: web/static/css/app.css (hunk 5 above)
Comment (martzoukos) [reply web/static/css/app.css]: Genau.

File: web/templates/article/index.html.eex (hunk 7 above)
Comment (nburkley) [reply web/templates/article/index.html.eex:35]: noted, thanks!

File: web/static/css/app.css (hunk 5 above)
Comment (martzoukos) [reply web/static/css/app.css]: The thing after the double dashes is a Modifier (the M in BEM), so you would normally have base styles under `.preferences__input` and a style modifier under `.preferences__input--inline`.

**Comment by nburkley** [comment]
Thanks @martzoukos. Next project you're on BE, I'm on FE 🙀

**Comment by martzoukos** [comment]
I don't get this joke. I'm already a BE dev. 😎

**Comment by nburkley** [comment]
who said I was joking?

**Comment by martzoukos** [comment]
#fullstack

**Comment by nburkley** [comment]
@martzoukos Is that a bit better? If so I'll just go ahead and merge and @eschaefer can make a fresh branch from `develop` on Monday

**Comment by martzoukos** [comment]
Yup! 👍

**Comment by nburkley** [comment]
Thanks!

**Pull request merged**